cli/cmd/generate_azure.go

```diff
@@ -317,6 +317,7 @@ By default, this command will function interactively, prompting for the required
 )
 
 type AzureGenerateCommandExtraState struct {
+	AskAdvanced    bool
```

Comment:
Looks like this is done inconsistently (before and now).  GCP/Azure (with your change) both cache this, whereas AWS does not. I think it makes sense to cache this. If the customer answered advanced questions previously and is resuming the cache, probably makes sense to re-prompt them with everything they may have selected before.

Not sure if we should raise a different PR for that to not muddy the water here; but figured it was worth calling out.

Reply:
Good spot, I agree as we cache each of the advanced options.  I'll include in this PR.